cpp/src/lance/format/schema.cc

```diff
@@ -239,7 +254,15 @@ ::arrow::Result<std::shared_ptr<lance::encodings::Decoder>> Field::GetDecoder(
   }
 }
 
-std::shared_ptr<::arrow::Field> Field::ToArrow() const { return ::arrow::field(name(), type()); }
+std::shared_ptr<::arrow::Field> Field::ToArrow() {
+  if (is_extension_field()) {
+    auto ext_type = ::arrow::GetExtensionType(extension_name());
```

Comment:
How does this handle project?
For excample, if an image object has `<URI: str, data:binary>`. can it support `SELECT image.uri FROM dataset`?

I am ok if there is another issue to address that.

Reply:
let me add a test case to check

Reply:
so if `image` is of ImageType extension, projecting `image.uri` returns the storage schema. So in the returned schema it's got `image` as a struct and `uri` as the utf8 sub-field.